src/hotspot/share/gc/g1/g1FullGCCompactTask.cpp

```diff
@@ -91,6 +99,15 @@ void G1FullGCCompactTask::work(uint worker_id) {
     compact_region(*it);
   }
 
+  if (MarkSweepDeadRatio > 0) {
```

Comment:
This check is useless. If `MarkSweepDeadRatio == 0`, the `skipping_compaction_queue` is empty anyway. You could certainly assert that if `MarkSweepDeadRatio == 0`, then `skipping_compaction_queue` must be empty too.